### PR TITLE
Add Solution usestate_en.md

### PR DESCRIPTION
**Pull request by MRITUNJAY-SINGHH**
This README provides a user-friendly explanation of the behavior of a React application utilizing the useState() hook.

### CODE REVIEW

File: react-quiz/usestate_en.md [hunk 1]
```diff
@@ -1,4 +1,79 @@
...
+   - This re-render occurs because the state change prompts React to update the UI and reflect the new state.
+   - Console logs: "click", "render A"
+      ```jsx
+       // First Click
+      userEvent.click(screen.getByText('click me'));
+  
+   
+
+3. **Second Click:**
+   - Clicking the button again updates the state to true, triggering another re-render of the App component.
```
Comment (BFEdev) [inline react-quiz/usestate_en.md:42]: why this trigger another re-render even after state is already being set to true?

Comment (MRITUNJAY-SINGHH) [reply react-quiz/usestate_en.md:42]: ## React `setState` Behavior Explanation

In React, when you call the `setState` function, it doesn't immediately update the state and re-render the component. Instead, it schedules an update, and React will perform the actual state update and re-render at some point in the future. This is done to batch multiple state updates together for efficiency.

### Code Execution Breakdown:

1. **Initial render:** `_state` is initially `false`.
2. **First click:** `setState(true)` is called, and React schedules an update. The component is re-rendered with the new state, logging `render App` and `render A`.
3. **Second click:** `setState(true)` is called again, and React schedules another update. The component is re-rendered once more, logging `render App` and `render A` again.
4. **Third click:** Same as before, `setState(true)` is called, React schedules an update, and the component re-renders, logging `render App` and `render A`.

Even though the state is already `true` after the first click, subsequent calls to `setState` still trigger React to schedule updates and re-render the component. This is to ensure that the component reflects the latest state, allowing React to handle potential side effects and keep the UI in sync with the underlying state.

---

**Additional Explanation:**

In React, the setState function doesn't immediately update the state and trigger a re-render. Instead, it schedules an update, and React handles the actual state update and re-render asynchronously. This behavior is designed to batch multiple state updates for efficiency.

Even if the state is already set to true after the first click, subsequent calls to setState still trigger React to schedule updates and re-render the component. React ensures that the component reflects the most up-to-date state, handling potential side effects and keeping the UI in sync.

In summary, the asynchronous nature of state updates in React, coupled with the mechanism to batch updates, results in the observed re-rendering behavior even when the state is already true.

**Review by MRITUNJAY-SINGHH — COMMENTED** [review]
added your question solution
